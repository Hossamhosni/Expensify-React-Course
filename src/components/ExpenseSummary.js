import React from 'react';
import numeral from 'numeral';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpenseSummary = (props) => (
	<div>
  		<p>Viewing {props.count} expenses totalling {numeral(props.total / 100).format('$0,0.00')}</p>
 	</div>
);

const mapStateToProps = (state) => {
	return {
		count: selectExpenses(state.expenses, state.filters).length,
		total: selectExpensesTotal(selectExpenses(state.expenses, state.filters))
	};
};

export default connect(mapStateToProps)(ExpenseSummary);