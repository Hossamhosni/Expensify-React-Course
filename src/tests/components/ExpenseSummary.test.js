import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary } from '../../components/ExpenseSummary';
import expenses from '../fixtures/expenses';
import selectExpensesTotal from '../../selectors/expenses-total';

test('should render correctly', () => {
	const wrapper = shallow(<ExpenseSummary />);
	expect(wrapper).toMatchSnapshot();
});

test('should render with the correct amount', () => {
	const wrapper = shallow(<ExpenseSummary count={expenses.length} total={selectExpensesTotal(expenses)}/>);
	expect(wrapper).toMatchSnapshot();
});