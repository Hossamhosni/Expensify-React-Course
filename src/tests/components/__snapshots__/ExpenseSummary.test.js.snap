// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div>
  <p>
    Viewing 
     expenses totalling 
    $0.00
  </p>
</div>
`;

exports[`should render with the correct amount 1`] = `
<div>
  <p>
    Viewing 
    3
     expenses totalling 
    $1,141.92
  </p>
</div>
`;
